ironfish/src/blockchain/blockchain.ts

```diff
@@ -1292,6 +1290,11 @@ export class Blockchain {
       tx,
     )
 
+    for (const transaction of block.transactions) {
```

Comment:
If you have a block with transaction A which mints some supply and transaction B which tries to spend that supply, that block will be invalid. Transaction B will have a spend which requires a valid witness, which won't hold true.

If you have a block with one transaction that mints some supply *and* has an output to transfer some tokens to another address, that will technically work (since the output circuit does not require a witness) and the value balancing will take care of the difference. It should work effectively the same if you replace the mint with a spend (assuming the merkle path verification passes).

We can add a test for this.

Reply:
Sounds good. Bitcoin has very clear same block guarantees, that are well defined in their consensus model. It would be nice if ours, or lack of ours were encoded in tests. Thanks